build_support/run_clang_tidy.py

```diff
@@ -267,6 +275,17 @@ def main():
     files = [make_absolute(entry['file'], entry['directory'])
              for entry in database]
 
+    if args.only_diff:
```

Comment:
Add a quick short comment before this block to note:

- We added this code
- Why we added this code
- If you ever expect someone to modify this, or if this is porcelain (I don't remember what git commands are porcelain vs plumbing), a quick description of what this code does

1 line each or less is fine, nothing fancy. This is just to make sure that if people replace this with a new run_clang_tidy script in the future, they won't wonder why only_diff is gone.